server/pulp/plugins/conduits/repo_publish.py

```diff
@@ -52,17 +54,20 @@ def __str__(self):
 
     def last_publish(self):
         """
-        Returns the timestamp of the last time this repo was published,
-        regardless of the success or failure of the publish. If
-        the repo was never published, this call returns None.
+        Returns the timestamp of the last time this repo was published, regardless of the success
+        or failure of the publish. If the repo was never published, this call returns None.
 
-        @return: timestamp instance describing the last publish
-        @rtype:  datetime.datetime or None
+        :return: timestamp instance describing the last publish
+        :rtype:  datetime.datetime or None
+
+        :raises DistributorConduitException: if any errors occur
         """
         try:
-            repo_publish_manager = manager_factory.repo_publish_manager()
-            last = repo_publish_manager.last_publish(self.repo_id, self.distributor_id)
-            return last
+            collection = RepoDistributor.get_collection()
```

Comment:
This is not good, but it mimics current behavior. I chose not to address this with this PR to limit the scope, especially because this will need to be redone with the RepoDistributor conversion.